test/rfq_blockchain_utils_test.ts

```diff
@@ -0,0 +1,162 @@
+            erc20Artifacts.DummyERC20Token,
+            provider,
+            { from: taker, gas: 10000000 },
+            {},
+            'The token that originally belongs to the maker',
+            'takerToken',
+            new BigNumber(18),
+            new BigNumber(1000000),
+        );
+
+        makerAmount = new BigNumber(100);
+        takerAmount = new BigNumber(50);
+        invalidTakerAmount = new BigNumber(100);
+
+        zeroEx = await fullMigrateAsync(
+            owner,
+            provider,
+            { from: owner, gasPrice: GAS_PRICE },
+            {},
+            { protocolFeeMultiplier: Number(PROTOCOL_FEE_MULTIPLIER) },
+            {
+                nativeOrders: zeroExArtifacts.NativeOrdersFeature,
+                metaTransactions: zeroExArtifacts.MetaTransactionsFeature,
+            },
+        );
+
+        rfqOrder = new RfqOrder({
+            makerToken: makerToken.address,
+            takerToken: takerToken.address,
+            makerAmount,
+            takerAmount,
+            maker,
+            taker,
+            txOrigin,
+            expiry: VALID_EXPIRY,
+            salt: new BigNumber(1),
+            verifyingContract: zeroEx.address,
+            chainId: CHAIN_ID,
+        });
+        orderSig = await rfqOrder.getSignatureWithProviderAsync(provider);
+
+        unfillableRfqOrder = new RfqOrder({
+            makerToken: makerToken.address,
+            takerToken: takerToken.address,
+            makerAmount,
+            takerAmount: invalidTakerAmount,
+            maker,
+            taker,
+            txOrigin,
+            expiry: VALID_EXPIRY,
+            salt: new BigNumber(1),
+            verifyingContract: zeroEx.address,
+            chainId: CHAIN_ID,
+        });
+        sigForUnfillableOrder = await unfillableRfqOrder.getSignatureWithProviderAsync(provider);
+
+        await makerToken.mint(makerAmount).awaitTransactionSuccessAsync({ from: maker });
+        await makerToken
+            .approve(zeroEx.address, makerAmount)
+            .awaitTransactionSuccessAsync({ from: maker });
+        await takerToken.mint(takerAmount).awaitTransactionSuccessAsync({ from: taker });
+        await takerToken
+            .approve(zeroEx.address, takerAmount)
+            .awaitTransactionSuccessAsync({ from: taker });
+
+        rfqBlockchainUtils = new RfqBlockchainUtils(zeroEx);
+    });
+
+    after(async () => {
+        await teardownDependenciesAsync(SUITE_NAME);
+    });
+
+    describe('validateMetaTransaction', () => {
+        it('returns true for a valid metatransaction', async () => {
+            const metaTx = rfqBlockchainUtils.generateMetaTransaction(rfqOrder, orderSig, taker, takerAmount, CHAIN_ID);
+            const metaTxSig = await metaTx.getSignatureWithProviderAsync(provider);
+
+            expect(await rfqBlockchainUtils.validateMetaTransactionOrThrowAsync(metaTx, metaTxSig, txOrigin)).to.eq(true);
```

Comment:
It would be really interesting if you can take this test to the next level and actually asset that the swap was successful? Maybe by having `validateMetaTransactionOfThrowAsync` return the amount filled? Just an idea :)

Reply:
👍